public/app/features/alerting/unified/api/buildInfo.test.ts

```diff
@@ -102,64 +172,82 @@ describe('discoverDataSourceFeatures', () => {
       expect(mocks.fetchRules).toHaveBeenCalledTimes(1);
       expect(mocks.fetchRules).toHaveBeenCalledWith('Loki');
     });
-  });
-
-  describe('When buildinfo returns 404 error', () => {
-    it('Should return cortex with ruler API disabled when prom rules works and ruler api returns not avaiable errors', async () => {
-      fetch.mockReturnValue(
-        throwError(() => ({
-          status: 404,
-        }))
-      );
-
-      mocks.fetchTestRulerRulesGroup.mockRejectedValue({
-        status: 404,
-        data: {
-          message: 'page not found',
-        },
-      });
-      mocks.fetchRules.mockResolvedValue([]);
+    it('Should return Thanos with enabled ruler API', async () => {
+      fetch
+        .mockReturnValueOnce(
+          of({
+            data: {
+              status: 'success',
+              data: {
+                version: '0.28.0',
+              },
+            },
+          })
+          // For the second call (to thanos flags api) return flags as this is a mock thanos datasource
+        )
+        .mockReturnValueOnce(
+          of({
+            data: {
+              'grpc-address': 'true',
+            },
+          } as ThanosFlagsResponse)
+        )
+        // Third call to rules API should fail
+        .mockRejectedValueOnce(new Error('404'));
 
       const response = await discoverDataSourceFeatures({
         url: '/datasource/proxy',
-        name: 'Cortex',
+        name: 'Prometheus',
         type: 'prometheus',
       });
 
-      expect(response.application).toBe(PromApplication.Lotex);
-      expect(response.features.rulerApiEnabled).toBe(false);
-
-      expect(mocks.fetchTestRulerRulesGroup).toHaveBeenCalledTimes(1);
-      expect(mocks.fetchTestRulerRulesGroup).toHaveBeenCalledWith('Cortex');
-
-      expect(mocks.fetchRules).toHaveBeenCalledTimes(1);
-      expect(mocks.fetchRules).toHaveBeenCalledWith('Cortex');
+      expect(response.application).toBe(PromApplication.Thanos);
+      expect(response.features.rulerApiEnabled).toBe(true);
+      expect(mocks.fetchRules).not.toHaveBeenCalled();
+      expect(mocks.fetchTestRulerRulesGroup).toHaveBeenCalled();
     });
 
-    it('Should return cortex with ruler API enabled when prom rules works and ruler api returns cortex error', async () => {
-      fetch.mockReturnValue(
-        throwError(() => ({
-          status: 404,
-        }))
-      );
+    // Something in this test is breaking subsequent tests, so its got to run last...
```

Comment:
When running this test in isolation, it passes, but it seems to break subsequent tests unless literally run last in the test suite. I also had to redefine `fetch = jest.fn();` to get some of these to pass. There's obviously something I don't understand going on here, and it's making me worry!!!!!!

Reply:
Maybe it's because of the mutating of the fetch 🤦 , let's fix that